code/go/0chain.net/blobbercore/challenge/protocol.go

```diff
@@ -308,6 +308,7 @@ func (cr *ChallengeEntity) CommitChallenge(ctx context.Context, verifyOnly bool)
 		cr.Status = Cancelled
 		cr.StatusMessage = "challenge completion time expired"
 		cr.UpdatedAt = time.Now().UTC()
+		return common.NewError("challenge_time_expired", "")
```

Comment:
but we also have to save, no
if we throw error it'll not save also..
just in case if u save and return error.. it'll rollback

please change this with proper validation

Reply:
Yes that is true.
Thanks.